exp/lighthorizon/actions/accounts.go

```
@@ -0,0 +1,129 @@
+package actions
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/stellar/go/support/log"
+
+	"github.com/stellar/go/exp/lighthorizon/adapters"
+	"github.com/stellar/go/exp/lighthorizon/services"
+	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/go/protocols/horizon/operations"
+	"github.com/stellar/go/support/render/hal"
+	"github.com/stellar/go/toid"
+)
+
+const (
+	urlAccountId = "account_id"
+)
+
+func accountRequestParams(w http.ResponseWriter, r *http.Request) (string, pagination) {
+	var accountId string
+	var accountErr bool
+
+	if accountId, accountErr = getURLParam(r, urlAccountId); !accountErr {
+		sendErrorResponse(w, http.StatusBadRequest, "")
+		return "", pagination{}
+	}
+
+	paginate, err := paging(r)
+	if err != nil {
+		sendErrorResponse(w, http.StatusBadRequest, string(invalidPagingParameters))
+		return "", pagination{}
+	}
+
+	if paginate.Cursor < 1 {
+		paginate.Cursor = toid.New(1, 1, 1).ToInt64()
+	}
+
+	if paginate.Limit < 1 {
+		paginate.Limit = 10
```

This should probably result in `400 Bad Request`. If there's a bug in the client code it's better to return an error than change it to a correct value. At the same time we could set the value to default value if it's not set in query params.

---

updated in  https://github.com/stellar/go/pull/4453/commits/91fcd88715a810a201fd5b3ff6bcc517d1c88ee3